File: Cpp/PetShop.cpp
#include <bits/stdc++.h>
using namespace std;

class PetShop {
private:
    int id;
    string nama_produk;
    string kategori;
    double harga;

public:
    // Konstruktor Default
    PetShop() {
        this->id = 0;
        this->nama_produk = "";
        this->kategori = "";
        this->harga = 0.0;
    }

    // Konstruktor dengan Parameter
    PetShop(int id, string nama, string kategori, double harga) {
        this->id = id;
        this->nama_produk = nama;
        this->kategori = kategori;
        this->harga = harga;
    }

    // Getter dan Setter (dengan const pada getter)
    int get_id() const { return this->id; }
    void set_id(int id) { this->id = id; }
    string get_nama_produk() const { return this->nama_produk; }
    void set_nama_produk(string nama) { this->nama_produk = nama; }
    string get_kategori() const { return this->kategori; }
    void set_kategori(string kategori) { this->kategori = kategori; }
    double get_harga() const { return this->harga; }
    void set_harga(double harga) { this->harga = harga; }

    // Menampilkan data produk
    void display_info() const {
        cout << "ID: " << this->id << "\n";
        cout << "Nama Produk: " << this->nama_produk << "\n";
        cout << "Kategori: " << this->kategori << "\n";
        cout << "Harga: Rp " << fixed << setprecision(2) << this->harga << "\n";
        cout << "--------------------------\n";
    }

    // Destructor
    ~PetShop() {}
};

File: PHP/PetShop.php
<?php
// PetShop.php
// File ini mendefinisikan kelas dasar PetShop yang menyimpan data produk dasar.
// Atribut 'gambar' menyimpan foto produk dalam bentuk string base64.

class PetShop {
    // Deklarasi atribut kelas dengan visibility protected agar dapat diakses oleh kelas turunan
    protected $id;           // ID produk
    protected $nama_produk;  // Nama produk
    protected $kategori;     // Kategori produk
    protected $harga;        // Harga produk
    protected $gambar;       // Foto produk (dalam base64)

    // Konstruktor dengan parameter default
    public function __construct($id = 0, $nama_produk = "", $kategori = "", $harga = 0.0, $gambar = "") {
        $this->id = $id;                         // Set ID produk
        $this->nama_produk = $nama_produk;       // Set nama produk
        $this->kategori = $kategori;             // Set kategori produk
        $this->harga = $harga;                   // Set harga produk
        $this->gambar = $gambar;                 // Set foto produk (default kosong)
    }

    // Getter untuk id
    public function getId() {
        return $this->id; // Kembalikan nilai id
    }

    // Setter untuk id
    public function setId($id) {
        $this->id = $id; // Atur nilai id
    }

    // Getter untuk nama_produk
    public function getNamaProduk() {
        return $this->nama_produk; // Kembalikan nama produk
    }

    // Setter untuk nama_produk
    public function setNamaProduk($nama_produk) {
        $this->nama_produk = $nama_produk; // Atur nama produk
    }

    // Getter untuk kategori
    public function getKategori() {
        return $this->kategori; // Kembalikan kategori produk
    }

    // Setter untuk kategori
    public function setKategori($kategori) {
        $this->kategori = $kategori; // Atur kategori produk
    }

    // Getter untuk harga
    public function getHarga() {
        return $this->harga; // Kembalikan harga produk
    }

    // Setter untuk harga
    public function setHarga($harga) {
        $this->harga = $harga; // Atur harga produk
    }

    // Getter untuk gambar
    public function getGambar() {
        return $this->gambar; // Kembalikan foto produk
    }

    // Setter untuk gambar
    public function setGambar($gambar) {
        $this->gambar = $gambar; // Atur foto produk
    }
}
?>
